modules/tile-converter/src/3d-tiles-converter/helpers/b3dm-converter.ts

```diff
@@ -1,354 +1,7 @@
-
-    if (!material) {
-      material = {
-        alphaMode: 'OPAQUE',
-        doubleSided: false,
-        pbrMetallicRoughness: {
-          metallicFactor: 0,
-          roughnessFactor: 1
-        }
-      };
-
-      if (isTextureIndexExists) {
-        material.pbrMetallicRoughness.baseColorTexture = {
-          index: textureIndex,
-          texCoord: 0
-        };
-      } else {
-        material.pbrMetallicRoughness.baseColorFactor = [1, 1, 1, 1];
-      }
-
-      return material;
-    }
-
-    if (textureIndex !== null) {
-      material = this._setGLTFTexture(material, textureIndex);
-    }
-
-    return material;
-  }
-
-  /**
-   * Set texture properties in material with GLTF textureIndex
-   * @param {object} materialDefinition - i3s material definition
-   * @param {number} textureIndex - texture index in GLTF
-   * @returns {void}
-   */
-  _setGLTFTexture(materialDefinition, textureIndex) {
-    const material = {
-      ...materialDefinition,
-      pbrMetallicRoughness: {...materialDefinition.pbrMetallicRoughness}
-    };
-    // I3SLoader now support loading only one texture. This elseif sequence will assign this texture to one of
-    // properties defined in materialDefinition
-    if (
-      materialDefinition.pbrMetallicRoughness &&
-      materialDefinition.pbrMetallicRoughness.baseColorTexture
-    ) {
-      material.pbrMetallicRoughness.baseColorTexture = {
-        index: textureIndex,
-        texCoord: 0
-      };
-    } else if (materialDefinition.emissiveTexture) {
-      material.emissiveTexture = {
-        index: textureIndex,
-        texCoord: 0
-      };
-    } else if (
-      materialDefinition.pbrMetallicRoughness &&
-      materialDefinition.pbrMetallicRoughness.metallicRoughnessTexture
-    ) {
-      material.pbrMetallicRoughness.metallicRoughnessTexture = {
-        index: textureIndex,
-        texCoord: 0
-      };
-    } else if (materialDefinition.normalTexture) {
-      material.normalTexture = {
-        index: textureIndex,
-        texCoord: 0
-      };
-    } else if (materialDefinition.occlusionTexture) {
-      material.occlusionTexture = {
-        index: textureIndex,
-        texCoord: 0
-      };
-    }
-    return material;
-  }
-
-  /*
-   * Returns Features length based on attribute array in attribute object.
-   * @param {Object} attributes
-   * @returns {Number} Features length .
-   */
-  _getFeaturesLength(attributes) {
-    if (!attributes) {
-      return 0;
-    }
-    const firstKey = Object.keys(attributes)[0];
-    return firstKey ? attributes[firstKey].length : 0;
-  }
-
-  /* Checks that normals buffer is correct
-   * @param {TypedArray} normals
-   * @returns {boolean} true - normals are correct; false - normals are incorrect
-   */
-  _checkNormals(normals) {
-    // If all normals === 0, the resulting tileset is all in black colors on Cesium
-    return normals.find((value) => value);
+export default class B3dmConverter extends TilesConverter {
```

Comment:
Do we need these subclasses? They don't do much... Maybe just instantiate the TilesConverter with the different version numbers.